cnudie/iter.py

```diff
@@ -249,3 +249,169 @@ def iter_resources(
         component_filter=component_filter,
         reftype_filter=reftype_filter,
     )
+
+
+async def iter_async(
+    component: ocm.Component,
+    lookup: cnudie.retrieve.ComponentDescriptorLookupById=None,
```

Comment:
do we need an async-lookup (or should we at least encourage one)? -> I think there should be dedicated typedefs for async lookups (+ wrong lookups should be rejected)